src/granite_orm.cr

```diff
@@ -48,6 +48,39 @@ class Granite::ORM
     {% SETTINGS[:timestamps] = true %}
   end
 
+  # define getter and setter for parent relationship
+  macro belongs_to(model_name)
+    field {{model_name.id}}_id : Int64
+
+    # retrieve the parent relationship
+    def {{model_name.id}}
+      if parent = {{model_name.id.camelcase}}.find {{model_name.id}}_id
+        parent
+      else
+        {{model_name.id.camelcase}}.new
+      end
+    end
+
+    # set the parent relationship
+    def {{model_name.id}}=(parent)
+      @{{model_name.id}}_id = parent.id
+    end
+  end
+
+  # define getter for related children
+  macro has_many(children_table)
+    def {{children_table.id}}
+      {% children_class = children_table.id[0...-1].camelcase %}
```

Comment:
This seems like a very naive way to convert from plural to singular (via removing the last character, presumably an s). Crystal doesn't seem to have an inflection library yet, which seems like something we don't want to implement in an ORM. So maybe this is okay for now?